src/main/java/com/ibm/crypto/plus/provider/RSASignature.java

```diff
@@ -84,6 +103,23 @@ protected void engineInitSign(PrivateKey privateKey) throws InvalidKeyException
             throw new InvalidKeyException("Key is not an RSAPrivateKey");
         }
 
+        try {
+            if (provider.isFIPS()) {
```

Comment:
We could also do something like:
```
int[] specificModulesLen = null;
if (provider.isFIPS()) {
   specificModulesLen = RSAKeyFactory.ALLOWABLE_MODLEN_FIPS_SIGN;
}
RSAKeyFactory.checkKeyLengths(((java.security.interfaces.RSAPrivateKey) privateKey).getModulus().bitLength(),
                         RSAKeyGenParameterSpec.F4,
                         RSAKeyFactory.MIN_MODLEN_NONFIPS,
                         64 * 1024,
                         specificModulesLen);
```
And then in `RSAKeyFactory`, only do check if not `null`:
```
if (specificModulesLen != null && !Arrays.asList(specificModulesLen).contains(modulusLen)) {
    throw new InvalidKeyException("In FIPS mode, only 2048, 3072, or 4096 size of RSA key is accepted.");
}
```

Reply:
In FIPS mode, we have two lists actually, one for sign and one for verify.

Reply:
We can change to:
```
int[] specificModulesLen = null;
int minModulusLen = RSAKeyFactory.MIN_MODLEN_NONFIPS;
if (provider.isFIPS()) {
   specificModulesLen = RSAKeyFactory.ALLOWABLE_MODLEN_FIPS_SIGN;
   minModulusLen = RSAKeyFactory.MIN_MODLEN_FIPS_PUB;
}
RSAKeyFactory.checkKeyLengths(((java.security.interfaces.RSAPrivateKey) privateKey).getModulus().bitLength(),
                         RSAKeyGenParameterSpec.F4,
                         minModulusLen,
                         64 * 1024,
                         specificModulesLen);
```